botorch/acquisition/multi_objective/monte_carlo.py

```diff
@@ -752,3 +759,110 @@ def forward(self, X: Tensor) -> Tensor:
         return self._compute_qehvi(samples=samples, X=X) + self._prev_nehvi
+
+
+class MOMF(qExpectedHypervolumeImprovement):
+    def __init__(
+        self,
+        model: Model,
+        ref_point: Union[List[float], Tensor],
+        partitioning: NondominatedPartitioning,
+        sampler: Optional[MCSampler] = None,
+        objective: Optional[MCMultiOutputObjective] = None,
+        constraints: Optional[List[Callable[[Tensor], Tensor]]] = None,
+        X_pending: Optional[Tensor] = None,
+        cost_call: Callable[Tensor, Tensor] = None,
+        eta: float = 1e-3,
+        **kwargs: Any,
+    ) -> None:
+        r"""MOMF acquisition function supporting m>=2 outcomes.
+        The model needs to have train_obj that has a fidelity
+        objective appended to its end.
+        In the following example we consider a 2-D output space
+        but the ref_point is 3D because of fidelity objective.
+
+        See [Irshad2021MOMF]_ for details.
+
+        Example:
+            >>> model = SingleTaskGP(train_X, train_Y)
+            >>> ref_point = [0.0, 0.0, 0.0]
+            >>> cost_func = lambda X:5+X[...,-1]
+            >>> momf = MOMF(model, ref_point, partitioning,cost_func)
+            >>> momf_val = momf(test_X)
+
+        Args:
+            model: A fitted model.
+            ref_point: A list or tensor with `m+1` elements representing the reference
+                point (in the outcome space) w.r.t. to which compute the hypervolume.
+                This is a reference point for the objective values (i.e. after
+                applying`objective` to the samples).
+            partitioning: A `NondominatedPartitioning` module that provides the non-
+                dominated front and a partitioning of the non-dominated space in hyper-
+                rectangles. If constraints are present, this partitioning must only
+                include feasible points.
+            sampler: The sampler used to draw base samples. Defaults to
+                `SobolQMCNormalSampler(num_samples=128, collapse_batch_dims=True)`.
+            objective: The MCMultiOutputObjective under which the samples are evaluated.
+                Defaults to `IdentityMultiOutputObjective()`.
+            constraints: A list of callables, each mapping a Tensor of dimension
+                `sample_shape x batch-shape x q x m` to a Tensor of dimension
+                `sample_shape x batch-shape x q`, where negative values imply
+                feasibility. The acqusition function will compute expected feasible
+                hypervolume.
+            X_pending: A `batch_shape x m x d`-dim Tensor of `m` design points that have
+                points that have been submitted for function evaluation but have not yet
+                been evaluated. Concatenated into `X` upon forward call. Copied and set
+                to have no gradient.
+            cost_call: A callable cost function mapping a Tensor of dimension
```

Comment:
Let's document that the default assumption is that `s` is the last dimension of the input